docs/details/key-skew.md

```diff
@@ -1,43 +1,39 @@
 # Key/Partition Skew
 
-Key or partition skew is a frequent problem in Spark. Key skew can result in everything from slowly running jobs (with straglers), to failing jobs.
+Key or partition skew is a frequent problem in Spark. Key skew can result in everything from slowly running jobs (with straglers), to failing jobs. Bumping up memory could resolve an OOM error but does not solve the underlying problem.
 
 
-What is data skew?
-
-1. Usually caused during a transformation when the data in one partition ends up being a lot more than the others, bumping up memory could resolve an OOM error but does not solve the underlying problem
+What causes partition skew?
+Certain types of aggregations and windows can result in partitioning the data on a particular key.
 
-1. Processing partitions are unbalanced by a magnitude then the largest partition becomes the bottleneck
+1. Usually caused during a transformation when the data in one partition ends up being a lot larger than the others.
 
+2. If processing partitions are unbalanced by an order of magnitude, then the largest partition becomes the bottleneck.
```

Comment:
Is (2) intended to mean partitions created BEFORE a transformation? i.e. do we need both (1) and (2) here?